src/controllers/TalksController.php

```diff
@@ -123,37 +133,307 @@ public function postAction($request, $db) {
                     } else {
-                        throw new Exception("The comment could not be stored", 400);
+                        throw new Exception("The comment could not be stored",
+                            400);
                     }
                 case 'starred':
                     // the body of this request is completely irrelevant
                     // The logged in user *is* attending the talk.  Use DELETE to unattend
                     $talk_mapper = new TalkMapper($db, $request);
                     $talk_mapper->setUserStarred($talk_id, $request->user_id);
-                    header("Location: " . $request->base . $request->path_info, NULL, 201);
+                    header("Location: " . $request->base . $request->path_info,
+                        null, 201);
                     exit;
                 default:
                     throw new Exception("Operation not supported, sorry", 404);
             }
         } else {
-            throw new Exception("method not yet supported - sorry");
+            throw new Exception('This should never have happened, sorry', 500);
+        }
+    }
+
+    /**
+     * Create a new Talk.
+     *
+     * This method creates a new talk after being called via the URL
+     * "/events/<ID>/talks" or "/talks" with the event-id being part of the
+     * POST-values
+     *
+     * @param Request $request
+     * @param DB $db
+     *
+     * @throws Exception on different occasions
+     * @return array|bool
+     */
+    public function createTalkAction($request, $db)
+    {
+        // Set the event-ID either via URL or via POST-Data depending on how the
+        // method was called.
+        // When it's called via a POST to /talks/ the event-id is part of the
+        // POST-data, when it's called via POST to /events/<id>/talks/ the
+        // event-id is part of the URL but will be overwritten by POST-data
+        $event_id = null;
+        if ($request->url_elements[2] == 'events' && isset($request->url_elements[3])) {
+            $event_id = $request->url_elements[3];
+        }
+        $event_id = filter_var(
+            $request->getParameter('event_id', $event_id),
+            FILTER_SANITIZE_NUMBER_INT
+        );
+
+        $event_mapper = new EventMapper($db, $request);
+        if (! $event_mapper->thisUserHasAdminOn($event_id)) {
+            throw new Exception('You do not have permissions to add a talk to this event', 403);
+        }
+        $eventlist = $event_mapper->getEventById($event_id);
+        if (count($eventlist['events']) < 1) {
+            throw new Exception('Event not found', 404);
+        }
+        $event = $eventlist['events'][0];
+
+        // incoming data
+        $talk   = array();
+        $errors = array();
+
+        $talk['event_id'] = $event_id;
+        if (empty($talk['event_id'])) {
+            $errors[] = '"event_id" is a required field';
+        }
+
+        $talk['talk_title'] = filter_var(
```

Comment:
You could probably achieve the below a bit more elegantly with a "schema" of sorts and a loop, eg:

```php
$schema = [
    'event_id' => FILTER_SANITIZE_INT,
    'talk_title' => FILTER_SANITIZE_STRING,
    'talk_description' => FILTER_SANITIZE_STRING,
    'talk_title' => FILTER_SANITIZE_STRING,
    'type' => FILTER_SANITIZE_STRING
];

foreach ($schema as $key => $filter) {
    $talk[$key] = filter_var($request->getParameter($key), $filter);
    if (empty($talk[$key])) {
        $errors[] = "The '{$key}' field is required";
    }
}
```